### ISSUE TITLE
Similar scheduling rules get merged but the UI fails to display correctly

### ISSUE THREAD

**Issue by kwaio** [rancher/rancher#11951]
When setting up 2 similar rules with the same check but different values, e.g.  

- Should not be in a host with a service named ServiceA
- Should not be in a host with a service named ServiceB

Rancher seems to concatenate the 2 values in a single label :
```
labels": {
"io.rancher.scheduler.affinity:container_label_soft_ne": "io.rancher.stack_service.name=StarName/ServiceA,io.rancher.stack_service.name=StackName/ServiceB"
},
```
While I believe rancher reads them right (I hope the order of the rules is enforced), but fails to display 2 lines : 
![image](https://user-images.githubusercontent.com/5406429/36985767-6a235a02-2098-11e8-89d5-05c1bc4da6a0.png)

---
| Useful | Info |
| :-- | :-- |
|Versions|Rancher `v1.6.10` Cattle: `v0.183.22` UI: `v1.6.23` |
|Access|`openldap` `admin`|
|Orchestration|`Cattle`|
|Route|`service.new`|

**Comment by kwaio** [comment]
Actually, from my tests, the rules do not seem to be enforced.
I had service A (2ct) and B (1ct) :

My rules say 

1.  A should not be with A (auto-anti-affinity)
2.  A should not be with B (soft anti-affinity)

And rancher did schedules both A containers on the same host. Unless I misunderstand how it's supposed to work, the rules weren't enforced properly.

**Comment by kwaio** [comment]
Still happens with  : 

Rancher `v1.6.18`
Cattle `v0.183.52`
User Interface `v1.6.43`

**Comment by loganhz** [comment]
Fixed in v1.6.22
Please let us know in a comment if you can still reproduce the issue, and we'll reopen it
